hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -395,15 +395,21 @@ private synchronized boolean getNamespaceEnabledInformationFromServer(
     try {
       LOG.debug("Get root ACL status");
       getClient().getAclStatus(AbfsHttpConstants.ROOT_PATH, tracingContext);
+      // If getAcl succeeds, namespace is enabled.
       isNamespaceEnabled = Trilean.getTrilean(true);
     } catch (AbfsRestOperationException ex) {
-      // Get ACL status is a HEAD request, its response doesn't contain
-      // errorCode
+      // Get ACL status is a HEAD request, its response doesn't contain errorCode
       // So can only rely on its status code to determine its account type.
-      if (HttpURLConnection.HTTP_BAD_REQUEST != ex.getStatusCode()) {
+      if (HttpURLConnection.HTTP_BAD_REQUEST == ex.getStatusCode()) {
+        // If getAcl fails with 400, namespace is disabled.
+        isNamespaceEnabled = Trilean.getTrilean(false);
+      } else if (HttpURLConnection.HTTP_NOT_FOUND == ex.getStatusCode()) {
+        // If getAcl fails with 404, namespace is enabled.
```

Comment:
Please revisit the http 404 error handling. This can occur in different cases, even with misconfigured url, which can lead to false positive condition by falling back to namespace_enabled.

Reply:
+1 on this, would suggest we check the errorCode and not the statusCode. The errorCode here should be: "FilesystemNotFound" (https://learn.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/get-properties?view=rest-storageservices-datalakestoragegen2-2019-12-12#:~:text=404%20Not%20Found%2C%20FilesystemNotFound%2C%20%22The%20specified%20filesystem%20does%20not%20exist.%22).

Reply:
Hi, there was a change in production code design, Updated the PR description with new implementation.
Tried to cover these points in PR Description itself.

@rakeshadr in case of 404 coming due to any other issue, file system APIs will start failing and any inference made on account type won't be relevant any more.

@saxenapranav head calls seems not to have error codes. Anyways, dependency on 404 status code is now removed.